mongoengine/base/document.py

```diff
@@ -746,14 +735,15 @@ def merge_index_specs(index_specs, indices):
             if not indices:
                 return index_specs
 
-            spec_fields = [v['fields']
-                           for k, v in enumerate(index_specs)]
+            spec_fields = [idx['fields'] for idx in index_specs]
```

Comment:
I see. But in case `fields` is a list of the lists the hashing would fail... I guess, the performance isn't crucial here and we can return to the original approach.